tests/numerics/fv/test_ad_tpfa.py

```diff
@@ -0,0 +1,830 @@
+import pytest
+
+import numpy as np
+import scipy.sparse as sps
+
+import porepy as pp
+
+from porepy.applications.md_grids.model_geometries import (
+    SquareDomainOrthogonalFractures,
+)
+
+
+class UnitTestAdTpfaFlux(
+    pp.constitutive_laws.AdDarcyFlux, pp.fluid_mass_balance.SinglePhaseFlow
+):
+    """
+
+    Notes for debugging:
+        - Set the pressure to a constant value in the initial condition to test only the
+          constant permeability component.
+    """
+
+    def __init__(self, params):
+        super().__init__(params)
+        self._neumann_face = 4
+        self._neumann_flux = 1529
+        self._nonzero_dirichlet_face = 5
+        self._dirichlet_pressure = 1683
+
+    def initial_condition(self):
+        super().initial_condition()
+        for _, data in self.mdg.subdomains(return_data=True):
+            pp.set_solution_values(
+                name=self.pressure_variable,
+                values=np.array([2, 3], dtype=float),
+                data=data,
+                iterate_index=0,
+                time_step_index=0,
+            )
+
+    def discretize(self):
```

Comment:
Do we still need this trick after adding zero in the core?

Reply:
Nope.